### PR TITLE
build, commit: allow removing default identity labels using `--identity-labels=false`

### PR DESCRIPTION
**Pull request by flouthoc**
Allow end users to remove default identity labels if they want to.
Since there are instances where images can be reproduced across version
hence users must have option to suppress default labels.

Closes: https://github.com/containers/buildah/issues/3826

### CODE REVIEW

**Comment by rhatdan** [comment]
Could you change the option to --identify-labels with a default value of true.

Then user could type --identify-labels=false. Shorter to type. 
Or perhaps --idlabels=false.

**Comment by flouthoc** [comment]
@rhatdan Would using default `true` cause problem for cases where `podman remote / API` or other tools which directly call buildah and initialize option in structs, we will have to instrument all such occurrences  of code to set default `true` however as of now they will just not set the field and it will remain `false`.

**Comment by rhatdan** [comment]
In the remote protocol, it should be an optional bool and only set when the user sets it.  Otherwise it defaults to true.

File: define/build.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ type CommonBuildOptions struct {
 	CPUSetMems string
 	// HTTPProxy determines whether *_proxy env vars from the build host are passed into the container.
 	HTTPProxy bool
+	// IdentityLabel if set ensures that default `io.buildah.version` label is not applied to build image.
+	IdentityLabel types.OptionalBool
```
Comment (flouthoc) [inline define/build.go:33]: Makes IdentityLabel optional

**Comment by flouthoc** [comment]
@nalind @rhatdan PTAL

File: pkg/cli/common.go [hunk 2]
```diff
@@ -227,6 +228,7 @@ func GetBudFlags(flags *BudResults) pflag.FlagSet {
 		panic(fmt.Sprintf("error marking the pull-never flag as hidden: %v", err))
 	}
 	fs.BoolVarP(&flags.Quiet, "quiet", "q", false, "refrain from announcing build instructions and image read/write progress")
+	fs.BoolVar(&flags.IdentityLabel, "identity-label", true, "add default builder label (default true)")
```
Comment (TomSweeneyRedHat) [inline pkg/cli/common.go:231]:
```suggestion
	fs.BoolVar(&flags.IdentityLabel, "identity-label", true, "add default identity label (default true)")
```
At least to be consistent with the man page

File: pkg/parse/parse.go [hunk 3]
```diff
@@ -146,25 +147,26 @@ func CommonBuildOptionsFromFlagSet(flags *pflag.FlagSet, findFlagFunc func(name
...
+		HTTPProxy:     httpProxy,
+		Memory:        memoryLimit,
+		MemorySwap:    memorySwap,
+		NoHosts:       noHosts,
+		ShmSize:       findFlagFunc("shm-size").Value.String(),
+		IdentityLabel: types.NewOptionalBool(identityLabel),
+		Ulimit:        ulimit,
+		Volumes:       volumes,
+		Secrets:       secrets,
+		SSHSources:    sshsources,
```
Comment (TomSweeneyRedHat) [inline pkg/parse/parse.go:169]: Can we put this in alpha order?

**Comment by flouthoc** [comment]
@rhatdan @TomSweeneyRedHat @nalind PTAL

File: pkg/parse/parse.go (hunk 3 above)
Comment (nalind) [reply pkg/parse/parse.go:169]: Please don't.  It creates noise for `git annotate`.

Comment (flouthoc) [reply pkg/parse/parse.go:169]: @nalind Everything was already in order I just moved new label in its proper place.

**Comment by rhatdan** [comment]
LGTM

**Comment by flouthoc** [comment]
@nalind @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@giuseppe @vrothberg @saschagrunert PTAL

**Review by vrothberg — COMMENTED** [review]
/lgtm

**Pull request merged**